geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/BinConversionProcess.scala

```diff
@@ -0,0 +1,155 @@
+  description = "Converts a feature collection to binary format"
+)
+class BinConversionProcess extends VectorProcess with LazyLogging {
+
+  /**
+    * Converts an input feature collection to BIN format
+    *
+    * @param features input features
+    * @param geom geometry attribute to encode in the bin format, optional
+    *             will use default geometry if not provided
+    * @param dtg date attribute to encode in the bin format, optional
+    *            will use default date if not provided
+    * @param track track id attribute to encode in the bin format, optional
+    *              will use feature id if not provided
+    * @param label label attribute to encode in the bin format, optional
+    * @param axisOrder axis order to encode points, optional
+    * @return
+    */
+  @DescribeResult(description = "Encoded feature collection")
+  def execute(
+              @DescribeParameter(name = "features", description = "Input feature collection to query ")
+              features: SimpleFeatureCollection,
+              @DescribeParameter(name = "geom", description = "Geometry field to use for BIN records", min = 0)
+              geom: String,
+              @DescribeParameter(name = "dtg", description = "Date field to use for BIN records", min = 0)
+              dtg: String,
+              @DescribeParameter(name = "track", description = "Track field to use for BIN records", min = 0)
+              track: String,
+              @DescribeParameter(name = "label", description = "Label field to use for BIN records", min = 0)
+              label: String,
+              @DescribeParameter(name = "axisOrder", description = "Axis order - either latlon or lonlat", min = 0)
+              axisOrder: String
+             ): java.util.Iterator[Array[Byte]] = {
+
+    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+    logger.debug(s"Running BIN encoding for ${features.getClass.getName}")
+
+    val sft = features.getSchema
+
+    val geomField  = {
+      val name = Option(geom).getOrElse(sft.getGeomField)
+      val axis = Option(axisOrder).map {
+        case o if o.toLowerCase(Locale.US) == "latlon" => AxisOrder.LatLon
+        case o if o.toLowerCase(Locale.US) == "lonlat" => AxisOrder.LonLat
+        case o => throw new IllegalArgumentException(s"Invalid axis order '$o'. Valid values are 'latlon' and 'lonlat'")
+      }
+      Some(GeometryAttribute(name, axis.getOrElse(AxisOrder.LonLat))) // note: wps seems to always be lon first
+    }
+    val dtgField   = Option(dtg).orElse(sft.getDtgField)
+    val trackField = Option(track).orElse(sft.getBinTrackId).orElse(Some("id"))
+    val labelField = Option(label)
+
+    // validate inputs
+    (geomField.map(_.geom).toSeq ++ dtgField ++ trackField ++ labelField).foreach { attribute =>
+      if (attribute != "id" && sft.indexOf(attribute) == -1) {
+        throw new IllegalArgumentException(s"Attribute $attribute doesn't exist in $sft")
```

Comment:
right, you'd just have to make sure the fields you specify exist in the input collection